### ISSUE TITLE
Node doesn't shutdown cleanly while trying to connect to another member

### ISSUE THREAD

**Issue by cangencer** [hazelcast/hazelcast#13345]
Node doesn't shutdown cleanly if there's some pending attempts to connect to another node. The connection threads are left hanging.

Simple reproducer:
``` 
Config config = new Config();
config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
config.getNetworkConfig().getJoin().getTcpIpConfig().setEnabled(true).addMember("192.168.1.100");
HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(config);
hazelcastInstance.shutdown();
```

**Comment by mdogan** [comment]
@cangencer: I didn't get the issue. Reproducer above exits cleanly, without any hanging threads. Can you explain the steps to reproduce?

**Comment by cangencer** [comment]
I tried this in 3.10.5, didn't reproduce but it did reproduce when I tried it again with 3.10. So I think it must have been fixed sometime in between?

**Comment by mdogan** [comment]
Ah I see. I was trying with 3.11 (master branch). Now with 3.10.x, even with latest maintenance branch, I see handing Hazelcast threads. Most probably this was fixed in 3.11, maybe unintentionally.

**Comment by Holmistr** [comment]
Closing as it seems not reproducible anymore.
